tsconfig.json

```diff
@@ -3,7 +3,7 @@
 		"module": "Node16",
 		"moduleResolution": "Node16",
 		"target": "ES2022",
-		"lib": ["ES2022"],
+		"lib": ["ES2023"],
```

Comment:
question

Would it be safer to have something like

```js
declare global {
  interface Array<T> {
    findLastIndex(
      predicate: (value: T, index: number, obj: T[]) => unknown,
      thisArg?: any
    ): number;
    findLast<S extends T>(predicate: (value: T, index: number, array: T[]) => value is S, thisArg?: any): S | undefined;
    findLast(predicate: (value: T, index: number, array: T[]) => unknown, thisArg?: any): T | undefined;
  }
}
```

so that ppl don't use `toSorted`, `with`, etc.?

Reply:
Methods like `toSorted` are unsupported on Node.js 18, so the CI should fail even if not adding such custom types.

Reply:
related: microsoft/TypeScript#56796